src/zenml/environment.py

```diff
@@ -100,27 +101,26 @@ def in_docker() -> bool:
 
     @staticmethod
     def in_google_colab() -> bool:
-        """If the current python process is running in a Google Colab."""
+        """If the current Python process is running in a Google Colab."""
         return "COLAB_GPU" in os.environ
 
     @staticmethod
     def in_notebook() -> bool:
-        """If the current python process is running in a notebook."""
-        try:
+        """If the current Python process is running in a notebook."""
+        if is_installed("IPython"):
             from IPython import get_ipython  # type: ignore
 
-            if get_ipython() is None:
-                # IPython is installed but not running from a notebook
-                return False
-            else:
+            ipython_status = get_ipython().__class__.__name__
+            if ipython_status in [
```

Comment:
aren't we missing an else clause here?

Reply:
I don't think so. It'll just return `False` on the final line of the function, as far as I can reason it.